datadict/datadict-postgres.inc.php

```diff
@@ -311,18 +311,27 @@ function _recreate_copy_table($tabname,$dropflds,$tableflds,$tableoptions='')
 	{
 		if ($dropflds && !is_array($dropflds)) $dropflds = explode(',',$dropflds);
 		$copyflds = array();
-		foreach($this->MetaColumns($tabname) as $fld) {
+		foreach(($meta=$this->MetaColumns($tabname)) as $fld) {
 			if (!$dropflds || !in_array($fld->name,$dropflds)) {
 				// we need to explicit convert varchar to a number to be able to do an AlterColumn of a char column to a nummeric one
 				if (preg_match('/'.$fld->name.' (I|I2|I4|I8|N|F)/i',$tableflds,$matches) &&
 					in_array($fld->type,array('varchar','char','text','bytea'))) {
 					$copyflds[] = "to_number($fld->name,'S9999999999999D99')";
+				} elseif (preg_match('/'.$fld->name.' ([\w]+)/i',$tableflds,$matches) &&
```

Comment:
I may be nit-picking and over-engineering things, but we're doing nearly the same regex match twice in a row, it would be more efficient to run preg_match() only once and use the capturing pattern `$matches[1]` in the if statements' conditions.

Moreover, in `[\w]+`, the square brackets are redundant - `\w+` is enough.